spyder/widgets/collectionseditor.py

```diff
@@ -1418,11 +1426,21 @@ def oedit(self, key):
 
     def plot(self, key, funcname):
         """Plot item"""
-        data = self.source_model.get_data()
         import spyder.pyplot as plt
-        plt.figure()
-        getattr(plt, funcname)(data[key])
-        plt.show()
+        data = self.source_model.get_data()
+
+        if self.get_conf('mute_inline_plotting', section='plots'):
+            from IPython.core.pylabtools import print_figure
+            fig, ax = plt.subplots()
+            getattr(ax, funcname)(data[key])
+            png = print_figure(fig)
```

Comment:
Shouldn't we use the format selected by users in Preferences here? I mean, not only `png` but also `jpeg` or `svg`?

Reply:
I assume you are referring to the settings in Preferences > IPython Console > Graphics. I did not know about them, there are so many preferences! 

Yes, I can ensure that that setting is honoured, and also the other settings in the same box: the width, height, resolution and tightness. By the way, I don't see a way in which jpeg can be selected; am I missing something?

Reply:
Sorry, you're right. We only support png's and svg's.